R/case_weights.R

```diff
@@ -79,36 +80,42 @@ NULL
 #'
 #' @export
 #' @name case-weight-helpers
-get_case_weights <- function(info, .data) {
+get_case_weights <- function(info, .data, call = rlang::caller_env()) {
   wt_col <- info$variable[info$role == "case_weights" & !is.na(info$role)]
 
-
   if (length(wt_col) == 1) {
     res <- .data[[wt_col]]
     if (!is.numeric(res)) {
-      rlang::abort(
-        paste0(
-          "Column ", wt_col, " has a 'case_weights' role but is not numeric."
-        )
+      cli::cli_abort(
+        c(
+          "x" = "{.arg {wt_col}} has a {.code case_weights} role,\\
+                 but is not numeric.",
+          "i" = "{.arg {wt_col}} is {.obj_type_friendly {wt_col}}."
+          ),
+        call = call
       )
     }
   } else if (length(wt_col) == 0) {
     res <- NULL
   } else {
-    too_many_case_weights(length(wt_col))
+    too_many_case_weights(wt_col, call = call)
   }
 
   res
 }
 
 # ------------------------------------------------------------------------------
 
-too_many_case_weights <- function(n) {
-  rlang::abort(
-    paste0(
-      "There should only be a single column with the role 'case_weights'. ",
-      "In these data, there are ", n, " columns."
-    )
+too_many_case_weights <- function(x, call = rlang::caller_env()) {
+  n <- length(x)
+
+  cli::cli_abort(
+    c(
+      "!" = "There should only be a single column with the role \\
+      {.code case_weights}.",
+      "i" = "In these data, there are {n} column{?s}: {.var {x}}"
```

Comment:
```suggestion
      "i" = "{cli::qty(n)}In these data, there {?is/are} {n} column{?s}: {.var {x}}"
```

So that we correctly pluralize is/are. Could we add a test case for multiple bad case weights columns?

``` r
msg <- "{cli::qty(n)}In these data, there {?is/are} {n} column{?s}: {.var {x}}"

x <- c("hey", "there")
n <- length(x)

cli::cli_abort(msg)
#> Error:
#> ! In these data, there are 2 columns: `hey` and `there`
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(msg)
#>  2.   └─rlang::abort(...)

x <- c("hey")
n <- length(x)

cli::cli_abort(msg)
#> Error:
#> ! In these data, there is 1 column: `hey`
#> Backtrace:
#>     ▆
#>  1. └─cli::cli_abort(msg)
#>  2.   └─rlang::abort(...)
```

<sup>Created on 2023-10-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I would maybe suggest adding a period at the end of this, too, but not sure if there's actually a grammar argument there.

Reply:
I just realized that this function is only called for `n>1` so it should always be called with pluralize